### PR TITLE
renamed the component useResponsiveComponent to makeResponsive

### LINKED ISSUE #298: Rename "useResponsiveComponent" since it's not a hook
## What:

The named export `useResponsiveComponent` should be renamed.

## Why:

Exporting functions that start with `use` in React ecosystem implies it's a [React hook](https://reactjs.org/docs/hooks-intro.html). In this case it's ambiguous, since `useResponsiveComponent` is rather a high-order component than a hook.

### PR DESCRIPTION
**Pull request by ruhaise**
## Changes

- renamed the component useResponsiveComponent to makeResponsive

## Issues

- Closes #298 

## Release version

- [ ] internal (no release required)
- [ ] patch (bug fixes)
- [ ] minor (backward-compatible changes)
- [x] major (backward-incompatible, breaking changes)

## Contributor's checklist

- [x] My branch is up-to-date with the latest `master`

- [x] I ran `yarn verify` to see the build and tests pass

### CODE REVIEW

File: examples/components/WithResponsiveProps/withResponsiveProps.jsx [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import React from 'react'
 import styled from 'supported-styling-library'
-import { useResponsiveComponent } from 'atomic-layout'
+import { withResponsiveProps } from 'atomic-layout'
```
Comment (kettanaito) [inline examples/components/WithResponsiveProps/withResponsiveProps.jsx:3]: What do you think of naming it `makeResponsive()`?

File: examples/components/WithResponsiveProps/withResponsiveProps.test.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
-describe('useResponsiveComponent', () => {
+describe('withResponsiveProps', () => {
   before(() => {
-    cy.loadStory(['hooks'], ['useresponsivecomponent'])
+    cy.loadStory(['components', 'withResponsiveProps'], ['default-behavior'])
```
Comment (kettanaito) [inline examples/components/WithResponsiveProps/withResponsiveProps.test.js:3]: I believe this should be:

```js
['components', 'withresponsiveprops'], ['default-behavior']
```

Notice that the second member is all lowercase. Sorry for such non-intuitive setup, yet it's how Storybook serializes story names. Once you correct this the tests should pass.

File: examples/components/WithResponsiveProps/withResponsiveProps.jsx (hunk 1 above)
Comment (ruhaise) [reply examples/components/WithResponsiveProps/withResponsiveProps.jsx:3]: agree,  I think it makes more sense :)

File: examples/components/WithResponsiveProps/withResponsiveProps.test.js (hunk 2 above)
Comment (ruhaise) [reply examples/components/WithResponsiveProps/withResponsiveProps.test.js:3]: yep, thanks for the heads-up 👍

**Comment by kettanaito** [comment]
Please, if you update to the latest `master`, I've added a `src/utils` folder that should contain the `query()` function as of now:

https://github.com/kettanaito/atomic-layout/tree/662757a74ec65bd690e26827bc8064c5f9baf79e/packages/atomic-layout/src/utils

Let's add this `makeResponsive()` function to the `utils/` directory as well. The test suit example should also be adjusted to live here:

https://github.com/kettanaito/atomic-layout/tree/662757a74ec65bd690e26827bc8064c5f9baf79e/examples/utils

File: examples/index.js [hunk 3]
```diff
@@ -94,19 +94,27 @@ import UseViewportChange from './hooks/UseViewportChange'
...
 
 /**
  * Utilities
  */
-import QueryExample from './utils/Query'
+
+// MakeResponsive
+import MakeResponsive from './utils/MakeResponsive/MakeResponsive'
+
+storiesOf('Utilities|MakeResponsive', module).add(
```
Comment (kettanaito) [inline examples/index.js:111]: I believe that `makeResponsive` has the only behavior, so nesting test examples with "Default behavior" looks odd.

I recommend to structure them as:

```js
storiesOf('Utilities', module)
  .add('query', ...)
  .add('makeResponsive', ...)
```

**Comment by ruhaise** [comment]
@kettanaito  the docs needs to be updated after the release of this changes.

**Comment by kettanaito** [comment]
Thank you for the changes, @ruhaise!
I'll update the docs one this is released. You rock.

**Pull request merged**